src/mapping_igvc/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(mapping_igvc)
+
+add_definitions(--std=c++14)
+
+## Find catkin macros and libraries
+find_package(catkin REQUIRED COMPONENTS
+    roscpp
+    message_generation
+    std_msgs
+)
+
+################################################
+## Declare ROS messages, services and actions ##
+################################################
+
+## To declare and build messages, services or actions from within this
+## package, follow these steps:
+## * Let MSG_DEP_SET be the set of packages whose message types you use in
+##   your messages/services/actions (e.g. std_msgs, actionlib_msgs, ...).
+## * In the file package.xml:
+##   * add a build_depend tag for "message_generation"
+##   * add a build_depend and a run_depend tag for each package in MSG_DEP_SET
+##   * If MSG_DEP_SET isn't empty the following dependency has been pulled in
+##     but can be declared for certainty nonetheless:
+##     * add a run_depend tag for "message_runtime"
+## * In this file (CMakeLists.txt):
+##   * add "message_generation" and every package in MSG_DEP_SET to
+##     find_package(catkin REQUIRED COMPONENTS ...)
+##   * add "message_runtime" and every package in MSG_DEP_SET to
+##     catkin_package(CATKIN_DEPENDS ...)
+##   * uncomment the add_*_files sections below as needed
+##     and list every .msg/.srv/.action file to be processed
+##   * uncomment the generate_messages entry below
+##   * add every package in MSG_DEP_SET to generate_messages(DEPENDENCIES ...)
+
+## Generate messages in the 'msg' folder
+add_message_files(
+    DIRECTORY msg
+    FILES ConeObstacle.msg LineObstacle.msg Point2D.msg
```

Comment:
It looks like `LineObstacle.msg` is abstracting a piece-wise polynomial, implying that line obstacles are represented by single polynomial segments, and merging two line obstacles gives us a spline. So do we want a spline `msg` here as well?

Reply:
I considered it, but right now I don't see any need to publish the spines as messages. If all goes to plan, by the time they leave the obstacle manager, it's in the form of an occupancy grid.

Reply:
This makes sense 👍